daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -297,6 +305,20 @@ private[lf] final class Compiler(
   private[this] def compileCommands(cmds: ImmArray[Command]): t.SExpr =
     pipeline(translateCommands(Env.Empty, cmds))
 
+  private[this] def compileWithContractDisclosures(
+      sexpr: t.SExpr,
+      disclosures: ImmArray[DisclosedContract],
+  ): t.SExpr = {
+    val disclosureLambda = pipeline(
```

Comment:
The disclosure function has type `SUnit => SUnit`. Its side effects are to:
- prime the on ledger contract cache with disclosed contracts
- populate the disclosed contract key table with disclosed contracts